colcon_devtools/verb/version_check.py

```diff
@@ -31,16 +32,25 @@ def main(self, *, context):  # noqa: D102
         for entry_point in colcon_extension_points.values():
             distributions.add(entry_point.dist)
 
-        pypi = ServerProxy('https://pypi.python.org/pypi')
+        base_url = 'https://pypi.python.org/pypi/{project}/json'
         for dist in sorted(distributions, key=lambda d: d.project_name):
-            versions = pypi.package_releases(dist.project_name)
-            if not versions:
+            url = base_url.format(project=dist.project_name)
+            response = urllib.request.urlopen(url)
+
+            if response.status != 200:
```

Comment:
I don't think this is sufficient. The `urlopen()` call might raise in case of HTTP status codes 4xx.

Reply:
Added cases for `URLError` and `HTTPError` as outlined: https://docs.python.org/3/howto/urllib2.html#number-1

No internet case example:
```
colcon-argcomplete: Failed to reach server
colcon-bash: Failed to reach server
colcon-cd: Failed to reach server
colcon-cmake: Failed to reach server
colcon-core: Failed to reach server
colcon-defaults: Failed to reach server
colcon-devtools: Failed to reach server
```

Incorrect package name example:
```
colcon-argcomplet: Server could not fulfill the request
colcon-bash 0.4.2: up-to-date
colcon-cd 0.1.1: up-to-date
colcon-cmake 0.2.26: up-to-date
colcon-core 0.6.1: up-to-date
colcon-defaults 0.2.5: up-to-date
colcon-devtools 0.2.2: up-to-date
colcon-library-path 0.2.1: up-to-date
```